### PR TITLE
Implements Hive Input/Output format to use vineyard as the storage backend

### LINKED ISSUE #1420: Hive integration: vineyard serve as the storage backend for hive
Describe your problem
---------------------

- [x] Setup the dev-infra
   - [x] Document how to easily start/stop a hive server
   - [x] How to connect to the hive server for testing
- [x] Figure out how to add a custom format (i.e., vineyard)
   - https://cwiki.apache.org/confluence/display/Hive/FileFormats 
- [ ] Figure out how to add custom location protocol (i.e., vineyard)
   - not needed at the first stage, as the reader/writer in input/output format won't touch the underlying filesystem
   - the VineyardFileSystem could be extended as a proxy to communicate with external storage in the future.
- [x] Enable storing table data to vineyard
   - [ ] zero-serialization 
- [x] Enable retrieve table data from vineyard
   - [ ] zero-copy
- [ ] Connect the implementation with spark 
- [ ] Partitioned table support
- [ ] Distributed execution support

### PR DESCRIPTION
**Pull request by vegetableysm**
What do these changes do?
-------------------------

Support vineyard as a storage backend of hive.

Related issue number
--------------------

Fixes #1420

### CODE REVIEW

File: java/core/src/main/java/io/v6d/core/client/IPCClient.java [hunk 1]
```diff
@@ -289,30 +294,58 @@ public ClusterStatus getClusterStatus() throws VineyardException {
...
     }
 
+    // TODO: exception
+    public void sealBlob(ObjectID id) throws VineyardException {
+        val root = mapper.createObjectNode();
+        SealRequest.put(root, id);
+        this.doWrite(root);
+        val reply = new SealReply();
+        reply.get(this.doReadJson());
+    }
```
Comment (sighingnow) [inline java/core/src/main/java/io/v6d/core/client/IPCClient.java:337]: `BufferBuilder` itself already has a `seal()`, method.

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: java/hive/src/main/java/io/v6d/hive/ql/io/VineyardRowInputFormat.java [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        v2.allocateNew(5);
+        v2.setValueCount(5);
+        for (int i = 0; i < 5; i++) {
+            v1.set(i, i);
+            v2.set(i, i + 1);
+        }
+        return vectorSchemaRoot;
+    }
+
+    private void prepare() throws VineyardException{
```
[the remark this answers is not in the document]
Comment (vegetableysm) [reply java/hive/src/main/java/io/v6d/hive/ql/io/VineyardRowInputFormat.java]: Done

File: java/hive/src/main/java/io/v6d/hive/ql/io/VineyardRowInputFormat.java [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+            if (value.getVectorSchemaRoot() != null) {
+                System.out.println("Set vectorSchemaRoot succeed!");
+            }
+            tableNameValid = false;
+            return true;
+        }
+        return false;
+    }
+
+    private VectorSchemaRoot getSchemaRoot() {
```
[the remark this answers is not in the document]
Comment (vegetableysm) [reply java/hive/src/main/java/io/v6d/hive/ql/io/VineyardRowInputFormat.java]: Done

File: java/modules/basic/src/main/java/io/v6d/modules/basic/tensor/TensorBuilder.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
+package io.v6d.modules.basic.tensor;
```
[the remark this answers is not in the document]
Comment (vegetableysm) [reply java/modules/basic/src/main/java/io/v6d/modules/basic/tensor/TensorBuilder.java]: Get!

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: java/hive/README.rst [hunk 5]
```diff
@@ -215,24 +215,29 @@ Hive and Vineyard
...
         select * from hive_static_partition;
         select * from hive_static_partition where value=666;
         select * from hive_static_partition where value=114514;
 
-- Test static partition(BUG):
+- Test dynamic partition:
+
+    We must set the batch size to 1. Because ArrowColumnarBatchSerDe do not process dynamic partitioning. If
+    the batch size is not set to 1, ArrowColumnarBatchSerDe will batch data from different partitions together,
+    and the data will be written to the same partition, which will cause the data to be written incorrectly.
```
[the remark this answers is not in the document]
Comment (vegetableysm) [reply java/hive/README.rst]: fixed

**Comment by sighingnow** [comment]
Closing as in favour of #1520

**Pull request closed without merging**